Source/JavaScriptCore/inspector/protocol/DOM.json

```diff
@@ -695,6 +695,16 @@
             "parameters": [
                 { "name": "allow", "type": "boolean" }
             ]
+        },
+        {
+            "name": "getMediaStats",
+            "description": "Returns media stats for the selected node.",
+            "parameters": [
+                { "name": "nodeId", "$ref": "NodeId", "description": "Id of the node to retrieve mediastats for." }
+            ],
+            "returns": [
+                { "name": "attributes", "type": "object", "description": "An interleaved array of node attribute names and values." }
```

Comment:
A raw object doesn't make for a good protocol value because it isn't possible from the definition of the protocol to know even the general shape of this object without more information. We also won't be able to have future differences appear in the legacy protocol versions, since the contents of the attributes is hidden behind the opaque object.

For an example of a somewhat similar object that we have that is a collection of attributes and how the protocol type is used, see `ContextAttributes` in Canvas.json:42.

Reply:
JavaScript objects are inherently introspectible. The entire shape, much less the general one, can be inferred from the result.  So the justification for an explicit description of the type passed across the protocol boundary must be different than this. 

I also don't see anywhere within the Legacy protocol definitions where the types have changed across protocol versions. Can you point one out?

Reply:
I’ll dig up some when I’m at a computer tomorrow morning.